src/Workspaces/Remote/Core/Telemetry/ApiUsageIncrementalAnalyzerProvider.cs

```diff
@@ -0,0 +1,231 @@
+            {
+                lock (_reported)
+                {
+                    _reported.Remove(projectId);
+                }
+            }
+
+            public async Task AnalyzeProjectAsync(Project project, bool semanticsChanged, InvocationReasons reasons, CancellationToken cancellationToken)
+            {
+                lock (_reported)
+                {
+                    // to make sure that we don't report while solution load, we do this heuristic.
+                    // if the reason we are called is due to "document being added" to project, we wait for next analyze call.
+                    // also, we only report usage information per project once.
+                    // this telemetry will only let us know which API ever used, this doesn't care how often/many times an API
+                    // used. and this data is approximation not precise information. and we don't care much on how many times
+                    // APIs used in the same solution. we are rather more interested in number of solutions or users APIs are used.
+                    if (reasons.Contains(PredefinedInvocationReasons.DocumentAdded) ||
+                        _reported.Contains(project.Id))
+                    {
+                        return;
+                    }
+                }
+
+                var metadataSymbolUsed = new HashSet<ISymbol>();
+                foreach (var document in project.Documents)
+                {
+                    var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                    foreach (var operation in GetOperations(model, cancellationToken))
+                    {
+                        cancellationToken.ThrowIfCancellationRequested();
+
+                        if (metadataSymbolUsed.Count > Max)
+                        {
+                            // collect data up to max per project
+                            break;
+                        }
+
+                        // this only gather reference and method call symbols but not type being used.
+                        // if we want all types from metadata used, we need to add more cases 
+                        // which will make things more expansive.
+                        CollectApisUsed(operation, metadataSymbolUsed);
+                    }
+                }
+
+                var groupByAssembly = metadataSymbolUsed.GroupBy(s => s.ContainingAssembly);
+                var apiPerAssemlby = groupByAssembly.Select(g => new
+                {
+                    // mark all string as PII (customer data)
+                    AssemblyName = new TelemetryPiiProperty(g.Key.Identity.Name),
+                    AssemblyVersion = g.Key.Identity.Version.ToString(),
+                    Symbols = g.Select(s => new TelemetryPiiProperty(s.ToDisplayString(verboseFormat)))
```

Comment:
Please don't use the `DispayString()`. Instead, use `GetDocumentationCommentId()` because that's what we use in all our other API gathering tools and this allows us to correlate data sources.

Also, make sure to get the generic API. For example, if a project uses `IEnumerable<Customer>`, we want the doc id for `IEnumerable<T>`, not for the instantiated generic type. Same reason, otherwise generic APIs are impossible to count properly.

Reply:
Tagging @marklio

Reply:
yep. updated to document comment id. and it already do symbol.OriginalDefinition to get symbol shown in source or metadata